csrc/id_model/id_model.cpp

```diff
@@ -387,13 +422,15 @@ namespace {
 std::vector<std::vector<Val*>> getTriviallyMappedIds(Expr* expr) {
   std::vector<std::vector<Val*>> mapped_ids;
   if (auto merge = dynamic_cast<Merge*>(expr)) {
-    // Size-one domains should be broadcast, so just checking
-    // isBroadcast should be sufficient, but just in case if there's
-    // any missing conversion to broadcast
-    if (merge->inner()->isBroadcast() || merge->inner()->extent()->isOneInt()) {
+    // Note that broacast IDs may have extents larger than 1, thus
+    // merge->inner()->isBroadcast() is not a sufficient condition to
+    // check. Merging a non-broadcast ID with such a broadcast ID
+    // result in a non-broadcast ID with extent multiplied by the
+    // broadcast extent.
+    if (merge->inner()->extent()->isOneInt()) {
```

Comment:
Thanks a lot for pointing out the example in the description, I wasn't aware splitting a broadcast ID would produce two broadcast IDs.
`Split: bS2{1} by factor 4 -> bS3{1}, bS4{4}`